### ISSUE TITLE
Corrupted radiation

### ISSUE THREAD

**Issue by Plantain** [wrf-model/WRF#1198]
Hi,
I'm looking for help debugging what I suspect is a bug somewhere in RRTMG.

I'm seeing large blocks of clearly invalid radiation fields, see SWDOWN, SWDDIR.
![ex2](https://user-images.githubusercontent.com/319895/82462964-22642280-9ab4-11ea-8ee1-2eb5f7495bba.png)
![ex](https://user-images.githubusercontent.com/319895/82462966-23954f80-9ab4-11ea-8a7b-6791dce6a6ef.png)

Running with 40 processes and no OpenMP, the blocks look possibly like MPI tiles?

Has anyone seen anything like this before, or has ideas on what to investigate further to track this down?

WRF 4.2, icc 19.1, build config 66.
namelist is a slightly altered version of the published operational HRRR namelist:
&physics
 use_aero_icbc                       = .true.,
 aer_opt                             = 4,
 mp_physics                          = 8,     8,     3,
 ra_lw_physics                       = 4,     4,     1,
 ra_sw_physics                       = 4,     4,     1,
 radt                                = 10,    10,    30,
 CO2TF                               = 1,
 sf_sfclay_physics                   = 5,     5,     1,
 sf_surface_physics                  = 3,     3,     1,
 sf_lake_physics                     = 0,     0,     1
 bl_pbl_physics                      = 5,     5,     1,
 bldt                                = 0,     0,     0,
 bl_mynn_tkebudget                   = 0,
 bl_mynn_tkeadvect                   = .false.,
 bl_mynn_cloudpdf                    = 2,
 bl_mynn_edmf                        = 1,
 bl_mynn_edmf_mom                    = 0,
 bl_mynn_edmf_tke                    = 0,
 bl_mynn_mixlength                   = 2,
 grav_settling                       = 0,
 scalar_pblmix                       = 1,
 cu_physics                          = 0,     0,     0,
 cudt                                = 0,     0,     5,
 isfflx                              = 1,
 ifsnow                              = 1,
 icloud                              = 1,
 icloud_bl                           = 1,
 surface_input_source                = 1,
 num_soil_layers                     = 9,
 sf_urban_physics                    = 0,
 mp_zero_out                         = 2,
 mp_zero_out_thresh                  = 1.e-12,
 maxiens                             = 1,
 maxens                              = 3,
 maxens2                             = 3,
 maxens3                             = 16,
 ensdim                              = 144,
 usemonalb                           = .true.,
 rdlai2d                             = .true.,
 num_land_cat                        = 21,
 mosaic_lu                           = 1
 mosaic_soil                         = 1
 fractional_seaice                   = 0,
 seaice_threshold                    = 271.4,
 prec_acc_dt                         = 30.,
 swint_opt = 1,
/

**Comment by dudhia** [comment]
I would note aer_opt=4 may be a special choice, so try aer_opt=1 to
see if that narrows down the problem to this option.
Jimy

**Comment by Plantain** [comment]
That does look quite plausible, I will retest now. I got that recommended setting from the WRFSolar Users Guide, but it looks like aer_opt=4 hasn't made it into WRF-master yet(?).
if ( aer_opt.eq.0 .or. aer_opt.eq.2 .or. aer_opt.eq.3) then
      iaer = 10
else if ( aer_opt .eq. 1 ) then
      iaer = 6
endif
So iaer may be undefined, to unknown consequence.

Notably it only seems to affect shortwave charts, not longwave, which further supports your theory.

**Comment by dudhia** [comment]
Yes, that option is for special inputs. Not sure what happens if those
don't exist.

**Comment by Plantain** [comment]
Yep, looks like that was it. Shall I leave this open for a patch to exit() on an invalid aer_opt, or is it assumed a namelist must be valid? Silent corruption is quite a footgun.

![ex](https://user-images.githubusercontent.com/319895/82474206-9b6a7680-9ac2-11ea-832f-3f9c390da149.png)

**Comment by davegill** [comment]
@Plantain @dudhia @weiwangncar 

Yes, leave this issue open, we will explicitly close it with a PR.
Please rename the title of the issue to the exact thing that is being fixed (for example, something related to an undefined aer_opt value).

Thanks much for tracking this down.

**Comment by dudhia** [comment]
By the way, the WRF-Solar doc page is about to be updated. This was in the
works,
but this is more reason for it.
